src/routes/safe/components/Transactions/helpers/TxParametersDetail/index.tsx

```diff
@@ -53,19 +54,23 @@ export const TxParametersDetail = ({
   isTransactionExecution,
   isOffChainSignature,
 }: Props): ReactElement | null => {
-  const { nonce } = useSelector(currentSafe)
   const threshold = useSelector(currentSafeThreshold) || 1
   const defaultParameterStatus = isOffChainSignature && threshold > 1 ? 'ETH_HIDDEN' : 'ENABLED'
 
+  const [isAccordionExpanded, setIsAccordionExpanded] = useState(false)
+
   const { safeNonce = '' } = txParameters
-  const isSafeNonceFuture = parseInt(safeNonce, 10) > nonce
-  const [isAccordionExpanded, setIsAccordionExpanded] = useState(isSafeNonceFuture)
+  const safeNonceNumber = parseInt(safeNonce, 10)
+  const { lastTxNonce } = useLastTxNonce()
+
+  const isTxNonceInOrder = lastTxNonce ? safeNonceNumber === lastTxNonce + 1 : false
```

Comment:
```suggestion
  const isTxNonceInOrder = lastTxNonce == null ? safeNonceNumber === lastTxNonce + 1 : false
```

Reply:
Is it missing something in the suggestion?

Reply:
My point was that if `lastTxNonce` is 0, your check won't work.

Reply:
Thank you! Good catch.